amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/PollingConfig.java

```diff
@@ -35,13 +36,23 @@
 
     public static final Duration DEFAULT_REQUEST_TIMEOUT = Duration.ofSeconds(30);
 
+    /**
+     * Configurable functional interface to override the existing DataFetcher.
+     */
+    Function<DataFetcherProviderConfig, DataFetcher> dataFetcherProvider;
     /**
      * Name of the Kinesis stream.
      *
      * @return String
      */
-    @NonNull
-    private final String streamName;
+    private String streamName;
+
+    /**
+     * @param kinesisClient Client used to access Kinesis services.
+     */
+    public PollingConfig(KinesisAsyncClient kinesisClient) {
```

Comment:
Given that we have explicit constructors, we need to remove lombok's @RequiredArgsConstructor from @Data (@Data - @RequiredArgsConstructor) = @Getter, @Setter, @ToString, @EqualsAndHashCode

Reply:
Looks like adding in explicit constructors removes the RequiredArgsConstructor - tested this + https://github.com/rzwitserloot/lombok/issues/1269